packages/sign-client/src/controllers/engine.ts

```diff
@@ -873,36 +894,54 @@ export class Engine extends IEngine {
       );
       throw new Error(message);
     }
-    const { pairingTopic, requiredNamespaces, relays } = params;
+    const { pairingTopic, requiredNamespaces, optionalNamespaces, sessionProperties, relays } =
+      params;
     if (!isUndefined(pairingTopic)) await this.isValidPairingTopic(pairingTopic);
 
-    // validate required namespaces only if they are defined
-    if (!isUndefined(requiredNamespaces) && isValidObject(requiredNamespaces) === 0) {
-      return;
-    }
-
-    const validRequiredNamespacesError = isValidRequiredNamespaces(requiredNamespaces, "connect()");
-    if (validRequiredNamespacesError) throw new Error(validRequiredNamespacesError.message);
     if (!isValidRelays(relays, true)) {
       const { message } = getInternalError("MISSING_OR_INVALID", `connect() relays: ${relays}`);
       throw new Error(message);
     }
+
+    // validate required namespaces only if they are defined
+    if (!isUndefined(requiredNamespaces) && isValidObject(requiredNamespaces) !== 0) {
```

Comment:
Does this leave open the possibility of validation passing if `requiredNamespaces` is simply undefined or doesn't pass the `isValidObject()` check, since we only run `validateNamespaces` based on this `if`?

Reply:
`isValidObject` will throw if `requiredNamespaces` is undefined